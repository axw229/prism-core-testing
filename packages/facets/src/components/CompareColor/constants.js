export const baseClass = 'color__comparison'
export const containerClass = `${baseClass}__container`
export const containerHeaderClass = `${baseClass}__container__header`
export const containerHeaderButtonClass = `${baseClass}__container__header__button`
export const wrapperClass = `${baseClass}__wrapper`
export const prevBtnWrapperClass = `${baseClass}__prev-btn__wrapper`
export const buttonsClass = `${baseClass}__buttons`
export const buttonsVisibleClass = `${baseClass}__buttons--visible`
export const nextBtnWrapperClass = `${baseClass}__next-btn__wrapper`
export const backgroundColorClass = `${baseClass}__background__color`
export const queueWrapperClass = `${baseClass}__queue__wrapper`
