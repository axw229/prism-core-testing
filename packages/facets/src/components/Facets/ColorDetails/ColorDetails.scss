$top-level-class-string: '.color-info';
$neutral-background-color: var(getThemeColor('white'));
$neutral-foreground-color: var(getThemeColor('dark'));
$strip-chip-size: 2rem;
$strip-chip-margin: $strip-chip-size * 0.1;
$width-thin: 0.125rem;
$width-small: 0.25rem;
$width-medium: 1rem;
$tab-set-padding: 0 0.25rem;
$tab-btn-height: strip-units(32px / 16px) * 1rem;
$details-tab-item-v-spacing: 0.75rem;
$details-tab-padding: 1rem;

.color-detail-view {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  position: relative;
  width: 100%;
  font-family: var(getVarName('typography.titleFontFamily'));

  @include respond-above(sm) {
    flex-direction: row;
    margin-bottom: $width-small;
    width: auto;
  }
}

.color-detail {
  &__info-wrapper {
    margin-bottom: $width-small;
    position: relative;
    width: 100%;
  }

  &__scene-wrapper {
    background: #eee;
    display: none;
    height: auto;
    margin-bottom: $width-small;
    position: relative;

    &--displayed {
      display: block;
    }
  }

  &__ctas {
    &--mobile {
      display: block;

      @include respond-above(sm) {
        display: none;
      }
    }

    &--desktop {
      display: none;

      @include respond-above(sm) {
        display: block;
      }
    }
  }

  @include respond-above(sm) {
    &__info-wrapper,
    &__scene-wrapper {
      margin: 0;
    }

    &__info-wrapper {
      height: 100%;
      left: 0;
      position: absolute;
      top: 0;
      width: 50%;
    }

    &__scene-wrapper {
      display: initial;
      order: 2;
      width: 50%;
    }
  }

  @include respond-above(md) {
    &__info-wrapper {
      width: 40%;
    }

    &__scene-wrapper {
      width: 60%;
    }
  }
}

#{$top-level-class-string} {
  background-color: $neutral-background-color;

  &-house-shaped {
    &__name-number {
      display: flex;
      flex-direction: column-reverse;
      padding: 0;
    }

    &__name {
      display: inline-block;
      font-family: var(getVarName('typography.bodyFontFamily'));
      font-style: italic;
      font-size: 1rem;
      line-height: 1.4rem;
    }

    &__number {
      display: block;
      font-size: 0.875rem;
      line-height: 1.4rem;
    }

    &__coord-color-name {
      font-style: italic;
      font-size: 0.8125rem;
    }

    &__coord-color-number {
      font-family: var(getVarName('typography.titleFontFamily'));
      font-size: 0.75rem;
    }
  }

  &__main-info {
    position: relative;
    padding: 2rem $width-medium;
    color: var(getThemeColor('black'));

    &--dark-color {
      color: var(getThemeColor('white'));
    }

    &--add {
      position: absolute;
      right: 50%;
      bottom: 50%;
      transform: translate(50%, 50%);
      z-index: 2;

      @include respond-below(sm) {
        right: 13%;
      }
    }
  }

  &__expanded-title {
    position: relative;
    width: 80%;
    z-index: 3;
    font-size: 1rem;
    line-height: 1.3;
  }

  &__number-name {
    padding: 0;
    margin: 0;
    line-height: inherit;
    font-size: 1.2em;
  }

  &__number {
    display: block;
    font-weight: 400;
    text-transform: uppercase;
    opacity: $color-near-opacity;
  }

  &__name {
    display: inline-block;
    font-weight: 600;
  }

  &__rack-location,
  &__type {
    font-size: 0.8em;
    font-weight: 400;
    margin: 0.25em 0 0 0;
    opacity: $color-near-opacity;
    padding: 0;
  }

  &__strip {
    border-bottom: $width-thin solid $neutral-background-color;
    border-top: $width-thin solid $neutral-background-color;
    display: none;
    height: 2rem;
    list-style: none;
    margin: 0 0 1rem;
    max-width: 26rem;

    @include respond-between(sm, md) {
      margin: 0;
    }
  }

  &__strip-color,
  &__strip-location {
    display: flex;
    flex: 1 1 auto;
    justify-content: center;
    align-items: center;
  }

  &__strip-color {
    border-right: $width-thin solid $neutral-background-color;
    position: relative;
  }

  &__strip-color-info {
    position: absolute;
    padding: 0.75rem 1rem;

    &--active {
      position: relative;
      padding: 0 0.5rem;
    }
  }

  &__strip-color-name {
    font-family: var(getVarName('typography.titleFontFamily'));

    &--dark-color {
      color: var(getThemeColor('white'));
    }
  }

  &__strip-location {
    align-items: center;
    background: #fff;
    display: flex;
    justify-content: center;
  }

  &__strip-location-name {
    color: var(getThemeColor('dark'));
    font-size: 0.9rem;
  }

  &__max-chip {
    display: none;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    transform: scale(0.6, 0.6);
    transform-origin: left top;
    transition: transform 0.35s;
    width: 65%;

    &--dark-color {
      color: var(getThemeColor('white'));
    }
  }

  &__max-chip--maximized {
    transform: scale(1, 1);
    width: 100%;
  }

  &__display-toggles-wrapper {
    display: none;

    @include respond-above(sm) {
      display: block;
      height: 1.5rem;
      position: absolute;
      right: calc(50% + 1rem);
      transition: right 0.35s;
      top: 1rem;
      width: 1.5rem;
      z-index: 9;

      &--chip-maximized {
        right: 1rem;
      }

      &--swatch-size {
        border: 1px solid #000;
        height: auto;
        width: auto;

        &#{$top-level-class-string}__display-toggles-wrapper--dark-color {
          border: 1px solid #fff;
          color: white;
        }
      }
    }
  }

  &__display-toggles-icon {
    height: 80%;
    width: 40%;

    &--scene#{$top-level-class-string}__display-toggles-icon.svg-inline--fa {
      height: 100%;
      width: 100%;
    }
  }

  &__display-toggle-button {
    height: 100%;
    width: 100%;

    &--dark-color {
      color: white;
    }
  }

  &__color-swatch-link {
    color: inherit;
    width: 100%;

    &:active,
    &:focus {
      @include browser-highlight($outline-size, $outline-style, $outline-color);
    }

    &:active,
    &:focus,
    &:hover {
      text-decoration: underline;

      span {
        text-decoration: underline;
      }
    }

    @include respond-above(md) {
      text-align: left;
    }
  }

  &__swatch-size-toggle-button {
    align-items: center;
    display: none;
    justify-content: space-around;
    height: 1.5rem;
    width: 2rem;
    transform: rotate(90deg);
    outline: none;
  }

  &__tab-list {
    border-top: $width-small solid $neutral-background-color;
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    list-style: none;
    text-align: center;

    @include respond-above(sm) {
      border-right: $width-small solid $neutral-background-color;
    }
  }

  &__tab {
    display: flex;
    align-items: center;
    justify-content: center;
    color: var(getThemeColor('dark'));
    cursor: pointer;
    font-size: 0.85rem;
    height: $tab-btn-height;
    overflow: hidden;
    padding: $tab-set-padding;
    text-align: center;
    border-right: $width-thin solid #fff;
    width: percentage(3/1);
    border-bottom: none;

    @include respond-above(sm) {
      width: auto;
      flex: auto;
    }

    &:last-child {
      border-right: none;
    }

    &--dark-color {
      color: white;
    }

    &.react-tabs__tab--selected {
      background-color: $neutral-background-color;
      border-color: $neutral-background-color;
      border-radius: 0;
      color: $neutral-foreground-color;
    }

    &:active,
    &:focus,
    &:hover {
      text-decoration: underline;
    }
  }

  &__tab-copy {
    font-size: 0.8em;
    font-weight: 600;
    line-height: 1;
    text-transform: uppercase;
  }

  &__tab-panel {
    background-color: $neutral-background-color;
    box-sizing: border-box;
    overflow: auto;

    &.react-tabs__tab-panel--selected {
      border-top: $width-thin solid $neutral-background-color;

      @include respond-above(sm) {
        height: calc(100% - #{$tab-btn-height});
        overflow: hidden;
      }
    }
  }

  &__coord-colors {
    display: flex;
    flex-direction: column;
    justify-content: stretch;
    margin: 0;
    list-style: none;
    height: auto;
    padding-top: $width-thin;

    @include respond-above(sm) {
      height: 100%;
    }

    @include respond-between(sm, md) {
      flex-direction: row;
      padding-right: $width-thin / 2;
    }

    @include respond-above(md) {
      margin-right: $width-small;
      flex-direction: column;
    }
  }

  &__coord-color {
    background: #bbc;
    border-bottom: $width-small solid $neutral-background-color;
    color: var(getThemeColor('black'));
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    font-size: 0.8rem;
    justify-content: center;
    line-height: 1.3;
    padding: $width-medium;

    @include respond-between(sm, md) {
      margin: 0 $width-thin 0 0;
      border: none;
      width: percentage(1/3);
      justify-content: flex-start;
      align-items: flex-start;
      padding: $width-medium / 2;
    }

    @include respond-above(md) {
      padding: 0 $width-medium;
    }

    &--dark-color {
      color: var(getThemeColor('white'));
    }

    &:last-child {
      margin-bottom: 0;
      border-bottom: 0;
    }
  }

  &__coord-color-number {
    font-size: 13px;
    font-weight: 400;
    opacity: $color-near-opacity;
    text-transform: uppercase;
    margin: 0;
  }

  &__coord-color-name {
    font-size: 13px;
    font-weight: 600;
    margin: 0;
  }

  &__similar-colors {
    background-color: $neutral-background-color;
    display: flex;
    flex-wrap: wrap;
    height: 100%;
    list-style: none;
    padding: 0;
    margin: 0 ($width-thin / -2);

    @include respond-above(sm) {
      margin: 0 ($width-thin * 1.5) 0 ($width-thin / -2);
    }
  }

  &__similar-color {
    border: ($width-thin / 2) solid $neutral-background-color;
    border-top-width: $width-thin;
    border-bottom-width: 0;
    display: block;
    height: 50%;
    width: percentage(1/6);
    position: relative;

    &::before {
      content: '';
      display: block;
      padding-top: 100%;
      width: 1px;
      height: 0;

      @include respond-above(sm) {
        display: none;
      }
    }
  }

  &__similar-color-info {
    display: block;
    height: 100%;
    transition: transform 0.3s, z-index 0.3s;
    width: 100%;
    z-index: 1;
    position: absolute;
    top: 0;
    left: 0;

    &--active {
      transform: scale(1.25);
      z-index: 200;

      .color-info__similar-color-info-wrapper {
        opacity: 1;
        transform: scale(0.8);
        transition: transform 0.3s, opacity 0.3s;
      }
    }
  }

  &__details-tab-wrapper {
    font-size: 0.8rem;
    line-height: 1.1;
    padding: $details-tab-padding $width-medium $details-tab-padding - $details-tab-item-v-spacing;

    .view-family-link {
      font-size: 0.8rem;
      color: #39c;
      text-transform: capitalize;
      text-decoration: none;
    }
  }

  &__chunk {
    margin: 0 0 $details-tab-item-v-spacing;
  }

  &__visual-specifications {
    display: flex;
    flex-direction: column;
    justify-content: stretch;
    margin: 0;

    @include respond-above(sm) {
      flex-direction: row;
    }
  }

  &__visual-specification {
    list-style: none;
    margin: 0 0.4rem 0.2rem 0;

    @include respond-above(sm) {
      margin-bottom: 0;
    }
  }

  &__family-link {
    display: block;
    margin: 1rem 0;
  }

  &__description-term {
    display: inline;
    font-weight: 600;
    margin-right: 0.1rem;
  }

  &__description-definition {
    display: inline;
    margin-right: 0.3rem;
  }

  @include respond-above(sm) {
    &__main-info {
      display: flex;
      flex-direction: column;
      height: 60%;
      justify-content: space-between;
      padding: $width-medium;
    }

    &__additional-info {
      height: 40%;

      .react-tabs {
        display: flex;
        flex-direction: column;
        height: 100%;
      }
    }

    &__max-chip {
      display: block;
    }

    &__display-toggles-wrapper--chip-maximized {
      right: 2rem;
    }

    &__strip {
      display: flex;
    }

    &__swatch-size-toggle-button {
      display: flex;

      &#{$top-level-class-string}__display-toggle-button--active {
        display: none;
      }

      &#{$top-level-class-string}__display-toggle-button--alt#{$top-level-class-string}__display-toggle-button--active {
        display: flex;
      }
    }

    &__display-toggle-button--alt {
      display: none;
    }
  }

  @include respond-above(md) {
    &__display-toggles-wrapper {
      right: calc(60% + 1rem);

      &--chip-maximized {
        right: 1rem;
      }
    }
  }

  @include respond-above(lg) {
    &__strip {
      max-width: 28rem;
    }

    &__strip-color-info--active {
      padding: 0 1rem;
    }
  }
}
