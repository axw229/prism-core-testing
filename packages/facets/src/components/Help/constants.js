export const baseClass = `cvw-help`
export const wrapper = `${baseClass}__wrapper`
export const tabsContainer = `${baseClass}__tabs-container`
export const contentWrapper = `${baseClass}__content-wrapper`
export const helpContent = `${baseClass}__help-content`
export const helpContentHide = `${helpContent}--hide`
export const contentHeader = `${baseClass}__content-header`
export const contentDetails = `${baseClass}__content-details`
export const subContent = `${baseClass}__subcontent`
export const subContentHeader = `${baseClass}__subContent-header`
export const subContentDetails = `${baseClass}__subContent-details`
export const helpIcons = `${baseClass}__help-icons`
export const helpImages = `${baseClass}__help-images`
export const helpImagesMobile = `${baseClass}__help-images-mobile`
export const iconWrap = `${baseClass}__icon-wrap`
export const iconWrapUndoRedo = `${baseClass}__icon-wrap-undoredo`
export const iconInfo = `${baseClass}__icon-info`
export const secondIcon = `${baseClass}__second-icon`
export const SCROLL_SPEED = 500
