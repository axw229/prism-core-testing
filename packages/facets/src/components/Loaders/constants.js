// @flow
// the 5 colors used by loaders in their default state; taken from SW CVW circa 2019
export const DEFAULT_COLORS = [
  '#00b9e4',
  '#69be28',
  '#ffb612',
  '#e17000',
  '#b71234'
]
// how many colors there are (this is sort of magic, so don't touch it)
export const NUM_COLORS = DEFAULT_COLORS.length
// how many generated colors are allowed to be dark
export const NUM_DARKS = Math.floor(NUM_COLORS / 2)
// how many generated colors should be light
export const NUM_LIGHTS = NUM_COLORS - NUM_DARKS
// default brightness step between generated colors
export const GENERATED_COLOR_BRIGHTNESS_STEP = 10
// % of brightness spectrum edges considered to be too dark or too light
export const GENERATED_COLOR_EDGE_PCT = 0.1
