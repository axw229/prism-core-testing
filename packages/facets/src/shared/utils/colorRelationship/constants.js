export const FULL = 360
export const HALF = FULL / 2
export const QUARTER = HALF / 2
export const NUM_CLUSTERS = 12
export const CLUSTER_SIZE = FULL / NUM_CLUSTERS
