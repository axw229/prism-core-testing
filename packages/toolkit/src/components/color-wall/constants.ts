export const MAX_SWATCH_SIZE = 50
export const BASE_SWATCH_SIZE = 15
export const OUTER_SPACING = 20
export const MIN_SCROLLER_HEIGHT = 200
export const MAX_SCROLLER_HEIGHT = MIN_SCROLLER_HEIGHT * 3

export const MIN_BASE_SIZE = 14
export const MAX_BASE_SIZE = 45
export const ZOOMED_BASE_SIZE = 50
export const ACTIVE_SWATCH_SIZE = 150
const BLOOMED_SWATCH_MIN_SIZE = 70
export const BLOOMED_SWATCH_SIZES = new Array(4).fill(null).map((v, i, arr) => {
  return BLOOMED_SWATCH_MIN_SIZE + ((ACTIVE_SWATCH_SIZE - BLOOMED_SWATCH_MIN_SIZE) / arr.length) * (arr.length - 1 - i)
})

export const SWATCH_WIDTH_WRAP_THRESHOLD = 20 // will prefer wrapped view if unwrapped view target swatch size falls below threshold value

export const TITLE_SIZE_RATIOS = {
  1: 0.4, // l1 should be styled regular weight
  2: 0.4, // l2 should be styled bold weight
  3: 0.5 // l3 should be styled bold weight, and is larger than l1 and l2
}

export const TITLE_SIZE_MIN = 12
export const TITLE_SIZE_MAX = 16
