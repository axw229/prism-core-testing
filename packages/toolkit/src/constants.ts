export enum SCENE_TYPES {
  OBJECT = 'objects',
  ROOM = 'rooms',
  AUTOMOTIVE = 'automotive',
  FAST_MASK = 'fastmask'
}

export enum DRAG_TYPES {
  SWATCH = 'SWATCH'
}

export enum SCENE_VARIANTS {
  DAY = 'day',
  NIGHT = 'night',
  MAIN = 'main'
}

// ERRORS
export const SYSTEM_ERROR = 'SYSTEM_ERROR'

export enum BUTTON_POSITIONS {
  TOP = 'top',
  BOTTOM = 'bottom'
}

export const KEY_CODES = {
  KEY_CODE_ARROW_DOWN: 40,
  KEY_CODE_ARROW_LEFT: 37,
  KEY_CODE_ARROW_RIGHT: 39,
  KEY_CODE_ARROW_UP: 38,
  KEY_CODE_ENTER: 13,
  KEY_CODE_SPACE: 32,
  KEY_CODE_TAB: 9
}
