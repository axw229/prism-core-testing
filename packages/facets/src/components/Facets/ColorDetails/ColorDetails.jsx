// @flow
import React, { useContext, useEffect, useState } from 'react'
import { FormattedMessage } from 'react-intl'
import { useDispatch } from 'react-redux'
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs'
import { cloneDeep } from 'lodash/lang'
import * as GA from 'src/analytics/GoogleAnalytics'
import ColorWallContext from 'src/components/Facets/ColorWall/ColorWallContext'
import HeroLoader from 'src/components/Loaders/HeroLoader/HeroLoader'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'
import ConfigurationContext, {
  type ConfigurationContextType
} from 'src/contexts/ConfigurationContext/ConfigurationContext'
import ColorDataWrapper from 'src/helpers/ColorDataWrapper/ColorDataWrapper'
import type { SceneStatus } from 'src/shared/types/Scene'
import 'src/providers/fontawesome/fontawesome'
import type { Color } from '../../../shared/types/Colors.js.flow'
import { toggleColorDetailsPage } from '../../../store/actions/scenes'
import { Content } from '../ColorWall/ColorSwatch/ColorSwatch'
import ColorChipMaximizer from './ColorChipMaximizer'
import { type ColorDetailsCTAData, ColorDetailsCTAs } from './ColorDetailsCTAs'
import ColorDetailsScenes from './ColorDetailsScenes'
import ColorInfo from './ColorInfo'
import ColorStrip from './ColorStrip'
import ColorViewer from './ColorViewer'
import CoordinatingColors from './CoordinatingColors'
import SimilarColors from './SimilarColors'
import 'src/scss/convenience/visually-hidden.scss'
import './ColorDetails.scss'

const baseClass = 'color-info'
const mainInfoClass = `${baseClass}__main-info`

type Props = {
  onColorChanged?: (Color) => void,
  onSceneChanged?: (SceneStatus) => void,
  onVariantChanged?: (string) => void,
  onColorChipToggled?: (boolean) => void,
  familyLink?: string,
  loading?: boolean,
  colorFromParent?: Color,
  initialVariantName?: string,
  callsToAction?: ColorDetailsCTAData[]
}

type ColorMessage = {
  ...Color,
  doNotBroadcast?: boolean
}

export const ColorDetails = ({
  onColorChanged,
  onSceneChanged,
  onVariantChanged,
  onColorChipToggled,
  familyLink,
  loading,
  colorFromParent = {},
  initialVariantName,
  callsToAction = []
}: Props) => {
  const dispatch = useDispatch()
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const [color, setColor] = useState<ColorMessage>(colorFromParent) // in some case we add an extra prop to a color obj that tells the pub method to not fire
  const [tabIndex, setTabIndex] = useState<number>(0)
  const [isMaximized, setMaximized] = useState(false)

  useEffect(() => {
    if (colorFromParent && colorFromParent.id !== color.id) {
      const _color = cloneDeep(colorFromParent)
      _color.doNotBroadcast = true
      setColor(_color)
    }
  }, [colorFromParent])

  useEffect(() => {
    dispatch(toggleColorDetailsPage())
    return () => {
      dispatch(toggleColorDetailsPage())
    }
  }, [])

  useEffect(() => {
    if (color) {
      GA.pageView(`color-detail/${color.brandKey} ${color.colorNumber} - ${color.name}`, GA_TRACKER_NAME_BRAND[brandId])
    }
    if (onColorChanged && !color?.doNotBroadcast) {
      onColorChanged(color)
    }
    // force tab change to first tab if there is no coordinating colors tab
    color.coordinatingColors || setTabIndex(0)
  }, [color])

  const AddColorBtn = ({ style }: { style?: $Shape<CSSStyleDeclaration> }) => {
    const { colorDetailsAddColor } = useContext<ConfigurationContextType>(ConfigurationContext)

    if (!colorDetailsAddColor) return null

    return (
      <div className={`${mainInfoClass}--add`} style={style}>
        <ColorWallContext.Provider value={{ displayAddButton: true }}>
          <Content msg='' color={color} isMaximized={isMaximized} style={{ position: 'relative' }} />
        </ColorWallContext.Provider>
      </div>
    )
  }

  if (loading) {
    return <HeroLoader />
  }

  return (
    <>
      <div className='color-detail-view'>
        <ColorChipMaximizer
          addColorBtn={(style) => <AddColorBtn style={style} isMaximized={isMaximized} />}
          color={color}
          isMaximized={isMaximized}
          setMaximized={setMaximized}
          onToggle={onColorChipToggled}
        />
        <div className={'color-detail__scene-wrapper color-detail__scene-wrapper--displayed'}>
          <ColorDetailsScenes color={color} isMaximized={isMaximized} intitialVariantName={initialVariantName} />
        </div>
        <div className='color-detail__info-wrapper'>
          <div
            className={`${mainInfoClass}${color.isDark ? ` ${mainInfoClass}--dark-color` : ''}`}
            style={{ backgroundColor: color.hex }}
          >
            <ColorViewer color={color} />
            <ColorStrip key={color.id} color={color} onColorChanged={setColor} />
            {!isMaximized ? <AddColorBtn /> : null}
          </div>

          {callsToAction?.length ? (
            <ColorDetailsCTAs className='color-detail__ctas color-detail__ctas--mobile' data={callsToAction} />
          ) : null}

          <div className={`${baseClass}__additional-info`}>
            <Tabs
              selectedIndex={tabIndex}
              onSelect={(index) => {
                const tabNames = ['View Coord Color Section', 'View Similar Color Section', 'View Color Info Section']
                GA.event(
                  { category: 'Color Detail', action: tabNames[index], label: tabNames[index] },
                  GA_TRACKER_NAME_BRAND[brandId]
                )
                setTabIndex(index)
              }}
            >
              <TabList className={`${baseClass}__tab-list`} style={{ backgroundColor: color.hex }}>
                {color?.coordinatingColors?.coord1ColorId && (
                  <Tab
                    className={`coordinating-colors-tab ${baseClass}__tab ${
                      color.isDark ? `${baseClass}__tab--dark-color` : ''
                    }`}
                  >
                    <div className={`${baseClass}__tab-copy`}>
                      <FormattedMessage id='COORDINATING_COLORS' />
                    </div>
                  </Tab>
                )}
                <Tab
                  className={`similar-colors-tab ${baseClass}__tab ${
                    color.isDark ? `${baseClass}__tab--dark-color` : ''
                  }`}
                >
                  <div className={`${baseClass}__tab-copy`}>
                    <FormattedMessage id='SIMILAR_COLORS' />
                  </div>
                </Tab>
                <Tab
                  className={`color-info-tab ${baseClass}__tab ${color.isDark ? `${baseClass}__tab--dark-color` : ''}`}
                >
                  <div className={`${baseClass}__tab-copy`}>
                    <FormattedMessage id='DETAILS' />
                  </div>
                </Tab>
              </TabList>
              {color?.coordinatingColors?.coord1ColorId && (
                <TabPanel className={`${baseClass}__tab-panel`}>
                  <CoordinatingColors color={color} onColorChanged={setColor} />
                </TabPanel>
              )}
              <TabPanel className={`${baseClass}__tab-panel`}>
                <SimilarColors color={color} onColorChanged={setColor} />
              </TabPanel>
              <TabPanel className={`${baseClass}__tab-panel color-info__tab-panel-details`}>
                <ColorInfo color={color} familyLink={familyLink} />
              </TabPanel>
            </Tabs>
          </div>
        </div>
      </div>

      {callsToAction?.length ? (
        <ColorDetailsCTAs className='color-detail__ctas color-detail__ctas--desktop' data={callsToAction} />
      ) : null}
    </>
  )
}

export default ColorDataWrapper(ColorDetails)
