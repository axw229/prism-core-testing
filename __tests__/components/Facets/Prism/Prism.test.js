/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { Route } from 'react-router'
import { Prism, RootRedirect } from 'src/components/Facets/Prism/Prism'
import { SceneManager } from 'src/components/SceneManager/SceneManager'
import ColorWallRouteComponent from 'src/components/Facets/ColorWall/ColorWallRouteComponent'
import { ROUTE_PARAMS, ROUTE_PARAM_NAMES } from 'constants/globals'
import ColorDetails from 'src/components/Facets/ColorDetails/ColorDetails'

const homeRoute = '/'
const activeRoute = '/active'
const prismNavWithRouterSelect = 'withRouter(PrismNav)'
const livePaletteConnectSelect = 'Connect(LivePalette)'

const colorWallBaseUrl = `/${ROUTE_PARAMS.ACTIVE}/${ROUTE_PARAMS.COLOR_WALL}`
const colorWallUrlPattern = `${colorWallBaseUrl}(/.*)?`
const activeColorIdSeoUrlPattern = `/${ROUTE_PARAMS.ACTIVE}/${ROUTE_PARAMS.COLOR}/:${ROUTE_PARAM_NAMES.COLOR_ID}/:${ROUTE_PARAM_NAMES.COLOR_SEO}`

const getPrism = (props) => {
  let defaultProps = {
  }

  let newProps = Object.assign({}, defaultProps, props)
  return shallow(<Prism {...newProps} />)
}

describe('Prism routes', () => {
  let prism
  let pathMap = {}

  beforeAll(() => {
    if (!prism) {
      prism = getPrism()
    }

    pathMap = prism.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props()
      pathMap[routeProps.path] = routeProps.component
      return pathMap
    }, {})
  })

  it('should match snapshot', () => {
    expect(prism).toMatchSnapshot()
  })

  it('should have 9 Route components', () => {
    expect(prism.find(Route)).toHaveLength(9)
  })

  it('should show RootRedirect component for route defined as homeRoute constant', () => {
    expect(pathMap[homeRoute]).toBe(RootRedirect)
  })

  it('should show SceneManager component for route defined as activeRoute constant', () => {
    expect(pathMap[activeRoute].WrappedComponent).toBe(SceneManager)
  })

  it('should show ColorWallRouteComponent for route defined as colorWallUrlPattern constant', () => {
    expect(pathMap[colorWallUrlPattern]).toBe(ColorWallRouteComponent)
  })

  it('should show ColorDetails component for route defined as activeColorIdSeoUrlPattern constant', () => {
    expect(pathMap[activeColorIdSeoUrlPattern]).toBe(ColorDetails)
  })

  it('should show component defined as livePaletteConnectSelect constant', () => {
    expect(prism.find(livePaletteConnectSelect).exists()).toBe(true)
  })
})
