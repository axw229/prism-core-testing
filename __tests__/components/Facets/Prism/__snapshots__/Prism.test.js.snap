// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prism routes should match snapshot 1`] = `
<Fragment>
  <withRouter(PrismNav) />
  <hr />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/active"
  />
  <Route
    component={[Function]}
    path="/active/color-wall(/.*)?"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/active/color/:colorId/:colorSEO"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/active/colors-from-image"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/search"
  />
  <Connect(LivePalette) />
</Fragment>
`;
